### PR TITLE
refactor: Convert Cell stories to csf format

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Updated Cell Base's const, stories, and test files. 
- Updated Cell Display's const, stories, and test files. 
- Updated Cell MultiSelect's const, stories, and test files. 
- Updated Cell Select's const, stories, and test files. 
- Updated Cell's const, stories, and test files. 
- Updated `CellVariants` to `CellVariant`

## **Screenshots/Recordings**
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 19
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 26
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 28
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 38
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 39
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 43
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 46
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 47
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 54
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 16 11 55

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Cells/Cell/README.md [hunk 1]
```diff
@@ -79,21 +79,54 @@ Default: false
...
+| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> | <span style="color:gray;font-size:14px">DEFAULT</span> |
+| :-------------------------------------------------- | :------------------------------------------------------ | :----------------------------------------------------- |
+| boolean                                             | No                                                      | false                                                  |
+
 ## Usage
 
 ```javascript
-// Change import path to relative path.
-import Cell from 'app/component-library/components/Cells/Cell';
-import { CellVariants} from 'app/component-library/components/Cells/Cell.types';
```
Comment (georgewrmarshall) [inline app/component-library/components/Cells/Cell/README.md:87]: I feel removing these imports hinders the value of the documentation? OR will they be automatically imported if this example code is pasted into a file?

File: app/component-library/components/Cells/Cell/README.md [hunk 2]
```diff
@@ -79,21 +79,54 @@ Default: false
...
+// Cell Display
+<Cell
+  variant={CellVariant.Display}
+  avatarProps={AVATAR_PROPS}
+  title={TITLE}
+  secondaryText={SECONDARY_TEXT}
+  tertiaryText={TERTIARY_TEXT}
```
Comment (georgewrmarshall) [inline app/component-library/components/Cells/Cell/README.md:100]: Do you think it's obvious what type of data should be passed for all constants here? Is it obvious what props I can use in avatarProps? If I search AVATAR_PROPS in the code base will I find an example? Some things to think about.

File: app/component-library/components/Select/Select/SelectItem/SelectItem.tsx [hunk 3]
```diff
@@ -34,6 +34,7 @@ const SelectItem: React.FC<SelectItemProps> = ({
       disabled={isDisabled}
       onPress={onPress}
       onLongPress={onLongPress}
+      {...props}
```
Comment (georgewrmarshall) [inline app/component-library/components/Select/Select/SelectItem/SelectItem.tsx:37]: Should we add a unit test to test that props can be passed down to this component?

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Some non blocking comments

File: app/component-library/components/Cells/Cell/README.md (hunk 1 above)
Comment (brianacnguyen) [reply app/component-library/components/Cells/Cell/README.md:87]: I feel like it's better not to have them and have the devs deal with the imports themselves. it's unnecessary noise imo

File: app/component-library/components/Cells/Cell/README.md (hunk 2 above)
Comment (brianacnguyen) [reply app/component-library/components/Cells/Cell/README.md:100]: yes. there is a SAMPLE_AVATAR_PROPS

File: app/component-library/components/Select/Select/SelectItem/SelectItem.tsx (hunk 3 above)
Comment (brianacnguyen) [reply app/component-library/components/Select/Select/SelectItem/SelectItem.tsx:37]: Yes will add this to components with ...props after the conversion is finished

**Review by georgewrmarshall — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**